pandas/io/pickle.py

```diff
@@ -1,25 +1,35 @@
 """ pickle compat """
 import pickle
+from typing import Any, Optional
 import warnings
 
+from pandas._typing import FilePathOrBuffer
 from pandas.compat import pickle_compat as pc
 
-from pandas.io.common import get_handle, stringify_path
+from pandas.io.common import get_filepath_or_buffer, get_handle
 
 
-def to_pickle(obj, path, compression="infer", protocol=pickle.HIGHEST_PROTOCOL):
+def to_pickle(
+    obj: Any,
+    filepath_or_buffer: FilePathOrBuffer,
+    compression: Optional[str] = "infer",
+    protocol: int = pickle.HIGHEST_PROTOCOL,
+):
     """
     Pickle (serialize) object to file.
 
     Parameters
     ----------
     obj : any object
         Any python object.
-    path : str
-        File path where the pickled object will be stored.
+    filepath_or_buffer : str, path object or file-like object
+        File path, URL, or buffer where the pickled object will be stored.
```

Comment:
can you add a versionchanged tag here (1.0), noting the URLs

Reply:
Added versionchanged tag for ver 1.0.0 to mention it accepts URLs.
> .. versionchanged:: 1.0.0
>            Accept URL. URL has to be of S3 or GCS.

Reply:
Added versionchanged tag for ver 1.0.0 to mention it accepts URLs

> ... versionchanged:: 1.0.0
> Accept URL. URL has to be of S3 or GCS.